concent_api/gatekeeper/views.py

```diff
@@ -40,102 +42,107 @@ def upload(request):
     try:
         url_validator(loaded_golem_message.storage_cluster_address)
     except ValidationError:
-        return gatekeeper_access_denied_response('storage_cluster_address is not a valid URL.')
+        return gatekeeper_access_denied_response('storage_cluster_address is not a valid URL.', path_to_file, loaded_golem_message.subtask_id, client_pub_key)
     if loaded_golem_message.storage_cluster_address != settings.STORAGE_CLUSTER_ADDRESS:
-        return gatekeeper_access_denied_response('Given storage_cluster_address is not defined in settings.')
+        return gatekeeper_access_denied_response('Given storage_cluster_address is not defined in settings.', path_to_file, loaded_golem_message.subtask_id, client_pub_key)
 
     # -CLIENT_PUBLIC_KEY
     if not isinstance(loaded_golem_message.authorized_client_public_key, bytes):
-        return gatekeeper_access_denied_response('Wrong type of authorized_client_public_key variable.')
+        return gatekeeper_access_denied_response('Wrong type of authorized_client_public_key variable.', path_to_file, loaded_golem_message.subtask_id, client_pub_key)
     client_public_key_base64 = (b64encode(loaded_golem_message.authorized_client_public_key)).decode('ascii')
-    if request.META['HTTP_CONCENT_CLIENT_PUBLIC_KEY'] != client_public_key_base64:
-        return gatekeeper_access_denied_response('authorized_client_public_key is different then Concent-Client-Public-Key header.')
+    if client_pub_key != client_public_key_base64:
+        return gatekeeper_access_denied_response('authorized_client_public_key is different then Concent-Client-Public-Key header.', path_to_file, loaded_golem_message.subtask_id, client_pub_key)
 
     # -OPERATION
     if request.method == 'POST' and loaded_golem_message.operation != 'upload':
-        return gatekeeper_access_denied_response('Wrong operation variable for this request method.')
+        return gatekeeper_access_denied_response('Wrong operation variable for this request method.', path_to_file, loaded_golem_message.subtask_id, client_pub_key)
     if request.method == 'GET' and loaded_golem_message.operation != 'download':
-        return gatekeeper_access_denied_response('Wrong operation variable for this request method.')
+        return gatekeeper_access_denied_response('Wrong operation variable for this request method.', path_to_file, loaded_golem_message.subtask_id, client_pub_key)
 
     # -FILES
     if not all(isinstance(file, dict) for file in loaded_golem_message.files):
-        return gatekeeper_access_denied_response('Wrong type of files variable.')
-    if request.method == 'POST':
-        path_to_file = request.get_full_path().partition(reverse('gatekeeper:upload'))[2]
-    if request.method == 'GET':
-        path_to_file = request.get_full_path().partition(reverse('gatekeeper:download'))[2]
+        return gatekeeper_access_denied_response('Wrong type of files variable.', path_to_file, loaded_golem_message.subtask_id, client_pub_key)
```

Comment:
I see that this code has been removed but there was a bug here. What if the method is neither `GET` nor `POST`. You should have either an `assert` stating that it's impossible or have code to handle other methods (possibly by returning an error).